lib/foreman_rh_cloud.rb

```diff
@@ -14,4 +16,57 @@ def self.authentication_url
   def self.verify_ssl_method
     @verify_ssl_method ||= ENV['SATELLITE_RH_CLOUD_URL'] ? OpenSSL::SSL::VERIFY_NONE : OpenSSL::SSL::VERIFY_PEER
   end
+
+  def self.http_proxy_string
+    ForemanRhCloud.proxy_setting
+  end
+
+  def self.transformed_http_proxy_string
+    ForemanRhCloud.transform_scheme(ForemanRhCloud.proxy_setting)
+  end
+
+  def self.proxy_setting
+    HttpProxy.default_global_content_proxy&.full_url ||
+    ForemanRhCloud.cdn_proxy ||
```

Comment:
You have to pass the logger all the way to "public" methods, if you want it to be overridden in the task:

```suggestion
  def self.http_proxy_string(logger: Foreman::Logging.logger('background'))
    ForemanRhCloud.proxy_setting(logger: logger)
  end

  def self.transformed_http_proxy_string(logger: Foreman::Logging.logger('background'))
    ForemanRhCloud.transform_scheme(ForemanRhCloud.proxy_setting(logger: logger))
  end

  def self.proxy_setting(logger: Foreman::Logging.logger('background'))
    HttpProxy.default_global_content_proxy&.full_url ||
    ForemanRhCloud.cdn_proxy(logger: logger) ||
```